bot/bot.py

```diff
@@ -197,6 +213,38 @@ def _remove_root_aliases(self, command: commands.Command) -> None:
         for alias in getattr(command, "root_aliases", ()):
             self.all_commands.pop(alias, None)
 
+    async def process_ext_not_found(self) -> None:
+        """
+        Process extensions which are in unloaded cache but not found on the bot.
+
+        Send a message to #dev-alerts with the extensions, and remove them from the cache,
+        """
+        await self.wait_until_guild_available()
+        cache_dict = await self.unloads_cache.to_dict()
+
+        unsuccesfully_unloaded = {
+            ext: link for ext, link in cache_dict.items() if ext not in self.successfully_unloaded
+        }
+
+        if not unsuccesfully_unloaded:
+            return
+
+        extensions_msg = "\n- ".join(f"`{ext}`: {msg}" for ext, msg in unsuccesfully_unloaded.items())
+
+        dev_alerts_channel = self.get_channel(constants.Channels.dev_alerts)
+        core_dev_role = self.get_guild(constants.Client.guild).get_role(constants.Roles.core_developers)
+        msg = (
+            f"\u26a0 {core_dev_role.mention}\n"
+            "The following extensions were found in the cache but not on the bot:\n\n"
+            f"- {extensions_msg}"
+            "\n\nClearing them from the cache."
+        )
+
+        for ext in unsuccesfully_unloaded:
+            await self.unloads_cache.delete(ext)
```

Comment:
What happens if you delete half of these then one of them fail? You won't send the message that you have cleared the cache because it errors. *That said it will send a sentry error and appear in the log I believe, but my point still stands*

Can you move this down until after you have sent the message?